src/java.base/share/classes/java/util/zip/ZipFile.java

```diff
@@ -1796,8 +1790,17 @@ private static int countCENHeaders(byte[] cen, int size) {
             return count;
         }
 
-        public RandomAccessFile getFile() {
-            return zfile;
+        public void beforeCheckpoint() {
+            synchronized (zfile) {
+                FileDescriptor fd = null;
+                try {
+                    fd = f.getFD();
```

Comment:
Could you pls check:
zfile.getFD()
?

Reply:
You're a hawkeye! I had an amended version all the time locally, and tested that. The difference is indeed in this line. Thanks! Fixed.